GrandCentralBoard/Widgets/Harvest/BillableDates.swift

```diff
@@ -0,0 +1,42 @@
+//
+//  BillableDates.swift
+//  GrandCentralBoard
+//
+//  Created by Maciek Grzybowski on 03.06.2016.
+//  Copyright © 2016 Oktawian Chojnacki. All rights reserved.
+//
+
+import Foundation
+
+
+final class BillableDates {
+
+    let dates: [NSDate]
+
+    init(referenceDate: NSDate, numberOfPreviousDays: Int, includeWeekends: Bool?) {
+        let calendar = NSCalendar(calendarIdentifier: NSCalendarIdentifierGregorian)!
+        let sunday = 1
+        let saturday = 7
+
+        var previousDays: [NSDate] = []
+        var date = referenceDate
+
+        while previousDays.count < numberOfPreviousDays {
+            date = date.dateDaysAgo(1)
+
+            if let includeWeekends = includeWeekends where includeWeekends == false {
+                let day = calendar.component(.Weekday, fromDate: date)
+                let isWeekend = (day == sunday || day == saturday)
```

Comment:
And here is a nice method to check if it is weekend:
https://developer.apple.com/library/mac/documentation/Cocoa/Reference/Foundation/Classes/NSCalendar_Class/#//apple_ref/occ/instm/NSCalendar/isDateInWeekend:

(and then you can use locale calendar, instead of gregorian)